### PR TITLE
Renaming to Smart Alerts

### PR DESCRIPTION
**Pull request by efvink**
Rename SmartSignalResultItem to SmartAlert
Delete SmartSignalResult.cs
(Not include renaming: SignalResultItem, ResultItem*, SmartSignal, SmartSignals, Signal...)

### CODE REVIEW

File: src/management/server/SmartSignalsManagementApi/EndpointsLogic/SignalResultApi.cs [hunk 1]
```diff
@@ -28,7 +28,7 @@ public class SignalResultApi : ISignalResultApi
         /// <summary>
         /// The event name in the Signal Result store (Application Insights) that contains the signal result
         /// </summary>
-        private const string EventName = "SmartSignalResult";
+        private const string EventName = "SmartAlerts";
```
Comment (EranGon) [inline src/management/server/SmartSignalsManagementApi/EndpointsLogic/SignalResultApi.cs]: >SmartAlerts [](start = 42, length = 11)

SmartAlert

**Comment by aviled** [comment]
> namespace SmartSignalRunnerChildProcess

This PR will break existing detectors, so you will need to update all detectors to use the new nuget created from this change and deploy each one.

---
Refers to: src/analysis/SmartSignalRunnerChildProcess/SmartSignalRunnerMain.cs:7 in 4d4b81c. [](commit_id = 4d4b81ccb0632cd8879a00d71f4ed5dbe95f6918, deletion_comment = False)

**Comment by aviled** [comment]
> {

Can you please change the solution file name as well?

---
Refers to: src/analysis/SmartSignalRunnerChildProcess/SmartSignalRunnerMain.cs:8 in 4d4b81c. [](commit_id = 4d4b81ccb0632cd8879a00d71f4ed5dbe95f6918, deletion_comment = False)

File: src/management/server/SmartSignalsManagementApi/EndpointsLogic/SignalResultApi.cs (hunk 1 above)
Comment (aviled) [reply src/management/server/SmartSignalsManagementApi/EndpointsLogic/SignalResultApi.cs]: if you change this then please don't forget to change in the publisher as well

---
In reply to: [170515166](https://github.com/liadtal/Smart-Alerts/pull/86#discussion_r170515166) [](ancestors = 170515166)

**Comment by EranGon** [comment]
> {

not in this change!

---
In reply to: [368424217](https://github.com/liadtal/Smart-Alerts/pull/86#issuecomment-368424217) [](ancestors = 368424217)

---
Refers to: src/analysis/SmartSignalRunnerChildProcess/SmartSignalRunnerMain.cs:8 in 4d4b81c. [](commit_id = 4d4b81ccb0632cd8879a00d71f4ed5dbe95f6918, deletion_comment = False)

File: src/analysis/SmartSignalsAnalysis/AnalysisDependenciesInjector.cs [hunk 2]
```diff
@@ -4,13 +4,13 @@
...
 
-namespace Microsoft.Azure.Monitoring.SmartSignals.Analysis
+namespace Microsoft.Azure.Monitoring.SmartAlerts.Appliance.Analysis
 {
-    using Microsoft.Azure.Monitoring.SmartSignals.Clients;
-    using Microsoft.Azure.Monitoring.SmartSignals.RuntimeShared.ChildProcess;
-    using Microsoft.Azure.Monitoring.SmartSignals.SignalLoader;
-    using Microsoft.Azure.Monitoring.SmartSignals.SignalResultPresentation;
-    using Microsoft.Azure.Monitoring.SmartSignals.Trace;
+    using Microsoft.Azure.Monitoring.SmartAlerts.Appliance.RuntimeShared.ChildProcess;
```
Comment (EranGon) [inline src/analysis/SmartSignalsAnalysis/AnalysisDependenciesInjector.cs]: >RuntimeShared [](start = 59, length = 13)

remove

File: src/analysis/SmartSignalsAnalysis/ISmartSignalRunner.cs [hunk 3]
```diff
@@ -4,13 +4,13 @@
...
 
-namespace Microsoft.Azure.Monitoring.SmartSignals.Analysis
+namespace Microsoft.Azure.Monitoring.SmartAlerts.Appliance.Analysis
 {
     using System.Collections.Generic;
     using System.Threading;
     using System.Threading.Tasks;
-    using Microsoft.Azure.Monitoring.SmartSignals.RuntimeShared;
-    using Microsoft.Azure.Monitoring.SmartSignals.SignalResultPresentation;
+    using Microsoft.Azure.Monitoring.SmartAlerts.Appliance.RuntimeShared;
```
Comment (EranGon) [inline src/analysis/SmartSignalsAnalysis/ISmartSignalRunner.cs]: >RuntimeShared [](start = 59, length = 13)

remove

File: src/analysis/SmartSignalsAnalysis/ISmartSignalRunner.cs [hunk 4]
```diff
@@ -4,13 +4,13 @@
...
+    using Microsoft.Azure.Monitoring.SmartAlerts.SignalResultPresentation;
```
Comment (EranGon) [inline src/analysis/SmartSignalsAnalysis/ISmartSignalRunner.cs]: >SignalResultPresentation [](start = 49, length = 24)

Presentation

File: src/analysis/SmartSignalsAnalysis/UnidentifiedSmartAlertResourceTypeException.cs [hunk 5]
```diff
@@ -1,39 +1,39 @@
...
     using System;
     using System.Runtime.Serialization;
 
     /// <summary>
-    /// This exception is thrown when a signals' result item resource type is not one of the types supported by the signal.
+    /// This exception is thrown when a signals' smart alert resource type is not one of the types supported by the signal.
     /// </summary>
     [Serializable]
-    public class UnidentifiedResultItemResourceTypeException : Exception
+    public class UnidentifiedSmartAlertResourceTypeException : Exception
```
Comment (EranGon) [inline src/analysis/SmartSignalsAnalysis/UnidentifiedSmartAlertResourceTypeException.cs]: >UnidentifiedSmartAlertResourceTypeException [](start = 17, length = 43)

UnidentifiedResourceTypeException

File: src/SampleSignal/HighRequestCountSignal.cs [hunk 6]
```diff
@@ -3,7 +3,7 @@
 //        Copyright (c) Microsoft Corporation.  All rights reserved.
 // </copyright>
 //-----------------------------------------------------------------------
-namespace Microsoft.Azure.Monitoring.SmartSignals.SampleSignal
+namespace Microsoft.Azure.Monitoring.SmartAlerts.Appliance.SampleSignal
```
Comment (EranGon) [inline src/SampleSignal/HighRequestCountSignal.cs]: >Appliance [](start = 49, length = 9)

this is not part of the appliance

File: src/SDK/SmartSignals.ProjectTemplate/MySmartAlert.cs [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+    /// Each smart alert instance contains both the detected issue's data and representation properties.
+    /// </summary>
+    public class MySmartAlert : SmartAlert
+    {
+    /// <summary>
+    /// Initializes a new instance of the <see cref="SmartAlert"/> class.
+    /// </summary>
+    /// <param name="title">The smart alert's title.</param>
+    /// <param name="resourceIdentifier">The resource identifier that this smart alert applies to.</param>
+    public MySmartAlert(string title, ResourceIdentifier resourceIdentifier) : base(title, resourceIdentifier)
```
Comment (EranGon) [inline src/SDK/SmartSignals.ProjectTemplate/MySmartAlert.cs]: indent

File: src/SDK/SmartSignals.ProjectTemplate/MySmartAlert.cs [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+    using Microsoft.Azure.Monitoring.SmartAlerts;
+
+    /// <summary>
+    /// The base class for representing a specific Smart Alert.
+    /// Each smart alert instance contains both the detected issue's data and representation properties.
+    /// </summary>
+    public class MySmartAlert : SmartAlert
+    {
+    /// <summary>
+    /// Initializes a new instance of the <see cref="SmartAlert"/> class.
```
Comment (EranGon) [inline src/SDK/SmartSignals.ProjectTemplate/MySmartAlert.cs]: MySmartAlert

File: src/SDK/SmartSignals.ProjectTemplate/MySmartAlert.cs [hunk 9]
```diff
@@ -0,0 +1,20 @@
+﻿namespace $safeprojectname$
+{
+    using Microsoft.Azure.Monitoring.SmartAlerts;
+
+    /// <summary>
+    /// The base class for representing a specific Smart Alert.
+    /// Each smart alert instance contains both the detected issue's data and representation properties.
```
Comment (EranGon) [inline src/SDK/SmartSignals.ProjectTemplate/MySmartAlert.cs]: wrong

File: src/SDK/SmartSignals.ProjectTemplate/MySmartAlertDetector.cs [hunk 10]
```diff
@@ -0,0 +1,33 @@
+﻿namespace $safeprojectname$
+{
+    using System.Collections.Generic;    
+    using System.Linq;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.Azure.Monitoring.SmartAlerts;
+
+    /// <summary>
+    /// The base class for representing Smart Alert Detector
```
Comment (EranGon) [inline src/SDK/SmartSignals.ProjectTemplate/MySmartAlertDetector.cs]: >The base class for representing Smart Alert Detector [](start = 8, length = 52)

wrong

**Comment by EranGon** [comment]
>     <Name>Smart Signal Project</Name>

why did you change the cs files and not the project name?

---
Refers to: src/SDK/SmartSignals.ProjectTemplate/SmartSignals.vstemplate:4 in ea454c6. [](commit_id = ea454c6e1a2c3f9d5b2fa5ed4fd34484165eead3, deletion_comment = False)

File: src/SDK/SmartSignalsSDK/SmartAlert.cs [hunk 11]
```diff
@@ -1,25 +1,25 @@
...
-namespace Microsoft.Azure.Monitoring.SmartSignals
+namespace Microsoft.Azure.Monitoring.SmartAlerts
 {
     using System;
 
     /// <summary>
-    /// The base class for representing a specific item in the Smart Signal result.
-    /// Each result item instance contains both the detected issue's data and representation properties.
+    /// The base class for representing a specific Smart Alert.
+    /// Each smart alert instance contains both the detected issue's data and representation properties.
```
Comment (EranGon) [inline src/SDK/SmartSignalsSDK/SmartAlert.cs]: >smart aler [](start = 13, length = 10)

Smart Alert - make sure it's capitalized all over the code

File: src/SDK/SmartSignalsShared/SignalResultPresentation/SmartAlertQueryRunInfo.cs [hunk 12]
```diff
@@ -1,25 +1,25 @@
...
 {
     using System.Collections.Generic;
     using Newtonsoft.Json;
 
     /// <summary>
-    /// This class provides information on how to run queries that are part of a <see cref="SmartSignalResultItemPresentation"/> object. 
+    /// This class provides information on how to run queries that are part of a <see cref="SmartAlertPresentation"/> object. 
     /// </summary>
-    public class SmartSignalResultItemQueryRunInfo
+    public class SmartAlertQueryRunInfo
```
Comment (EranGon) [inline src/SDK/SmartSignalsShared/SignalResultPresentation/SmartAlertQueryRunInfo.cs]: >SmartAlertQueryRunInfo [](start = 17, length = 22)

QueryRunInfo

File: src/SDK/SmartSignalsShared/SignalResultPresentation/InvalidSmartAlertPresentationException.cs [hunk 13]
```diff
@@ -1,40 +1,40 @@
 ﻿//-----------------------------------------------------------------------
-// <copyright file="InvalidSmartSignalResultItemPresentationException.cs" company="Microsoft Corporation">
+// <copyright file="InvalidSmartAlertPresentationException.cs" company="Microsoft Corporation">
 //        Copyright (c) Microsoft Corporation.  All rights reserved.
 // </copyright>
 //-----------------------------------------------------------------------
 
-namespace Microsoft.Azure.Monitoring.SmartSignals.SignalResultPresentation
+namespace Microsoft.Azure.Monitoring.SmartAlerts.SignalResultPresentation
```
Comment (EranGon) [inline src/SDK/SmartSignalsShared/SignalResultPresentation/InvalidSmartAlertPresentationException.cs]: >SignalResultPresentation [](start = 49, length = 24)

Presentation

**Comment by EranGon** [comment]
>     public static class ConfigurationReader

this shouldn't be in the current project - please open a bug to move

---
Refers to: src/SDK/SmartSignalsShared/Tools/ConfigurationReader.cs:16 in ea454c6. [](commit_id = ea454c6e1a2c3f9d5b2fa5ed4fd34484165eead3, deletion_comment = False)

File: src/SmartSignalsRuntimeShared/SmartSignalsRuntimeShared.csproj [hunk 14]
```diff
@@ -1,8 +1,8 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
     <TargetFramework>net461</TargetFramework>
-    <RootNamespace>Microsoft.Azure.Monitoring.SmartSignals.RuntimeShared</RootNamespace>
-    <AssemblyName>Microsoft.Azure.Monitoring.SmartSignals.RuntimeShared</AssemblyName>
+    <RootNamespace>Microsoft.Azure.Monitoring.SmartAlerts.Appliance.RuntimeShared</RootNamespace>
```
Comment (EranGon) [inline src/SmartSignalsRuntimeShared/SmartSignalsRuntimeShared.csproj]: >RuntimeShared [](start = 68, length = 13)

remove

**Comment by efvink** [comment]
> {

I will change it later 

---
In reply to: [368427890](https://github.com/liadtal/Smart-Alerts/pull/86#issuecomment-368427890) [](ancestors = 368427890,368424217)

---
Refers to: src/analysis/SmartSignalRunnerChildProcess/SmartSignalRunnerMain.cs:8 in 4d4b81c. [](commit_id = 4d4b81ccb0632cd8879a00d71f4ed5dbe95f6918, deletion_comment = False)

**Pull request closed without merging**